2996_reflection/reflection.md

```diff
@@ -4993,12 +4988,22 @@ consteval vector<info> members_of(info r);
 * a function,
 * a non-static data member,
 * a namespace, or
-* a namespace alias.
+* a namespace alias,
+
+and if its first declaration is within a definition of `$E$`.
 
 [Counterexamples of representable members include: injected class names, partial template specializations, friend declarations, and static assertions.]{.note}
 
+[#]{.pnum} A member `$M$` of a class or namespace is _members-of-visible_ from a point `$P$` if there exists a declaration `$D$` of `$M$` that is reachable from `$P$`, and either `$M$` is not TU-local or `$D$` is declared in the translation unit containing `$P$`.
```

Comment:
> , and either `$M$` is not TU-local or

Do we need to say that? As if it's TU local, wouldn't it be excluded from being part of the module (as otherwise the module would have an exposure)?

Reply:
These definitions are so convoluted - but I believe TU-local declarations _are_ nevertheless attached to the module. And oddly enough, they are also "reachable" in the strictest sense:

> A declaration _D_ is _reachable from_ a point _P_ if [...]
> - _D_ is not discarded ([module.global.frag]), appears in a translation unit that is reachable from _P_, and does not appear within a private-module-fragment.

So any TU-local definition attached to a named module in a TU visible from _P_ is in fact reachable (TU-local definitions of course don't need to be in the private-module-fragment).

(https://eel.is/c++draft/module.reach#3)